scripts/orchestra_methods.py

```diff
@@ -0,0 +1,319 @@
+                sorted_h5s = sorted_h5s + [h5.replace("/cromwell_root/", "gs://")]
+                break
+    # Add the location column
+    df["Location"] = sorted_h5s
+    # Save the csv, output it
+    df.to_csv("analysis.csv", index=False)
+
+
+def orchestra_filter(paths, masterCsv, sampleIds,transMap):
+    """Map the initial inputs to CellRanger Count.
+        Arguments:
+        paths- gsurls to every fastq directory
+        masterCsv- Csv File mapping samples, lanes and indices to bcl
+        sampleIds- list of every sample, not actually needed, just easier
+        transMap- map of reference name to gsurl of transcriptome files
+        Outputs-
+        paths.tsv- Cromwell mapping of sample to a comma seperated string of fastq gsurls
+        reference_list.tsv- Cromwell mapping of sample to genome
+        transcriptome_list.tsv- Cromwell mapping of sample to transcriptome file
+        chemistry.tsv- Cromwell mapping of sample to chemistry
+    """
+    # get a list of every fastq/sample directory using gsutil ls
+    fastqs = []
+    for f in paths:
+        fastqs = fastqs + check_output(["gsutil", "ls", f]).decode("utf-8").split("\n")
+    # open the files for writing
+    chemistry_list = open('chemistry.tsv', 'w+')
+    reference_list = open('reference.tsv', 'w+')
+    transcriptome_list = open('transcriptome.tsv', 'w+')
+    paths_list = open('paths.tsv', 'w+')
+    # open the masterCsv and transcriptome map csv
+    df = pd.read_csv(masterCsv)
+    tm = pd.read_csv(transMap)
+    # create the maps
+    for sample in sampleIds:
+        # Sample paths map
+        # a path matches if it ends with .../sample_id/
+        key = os.path.join(sample,"")
+        filter_paths = ",".join([path for path in fastqs if path.endswith(key)])
+        # Write to file
+        paths_list.write("%s\t%s\n" % (sample, filter_paths))
+        # Chemistry and Genome map
+        # Chemistry and Genome are the same across sample (assumed) so we can just use the first one we get
+        rows = df.loc[df["Sample"] == sample]
+        chemistry = rows["Chemistry"].tolist()[0]
+        # first get the genome for the sample
+        reference = rows["Reference"].tolist()[0]
+        # we have a map of reference names to gsurls
+        transcriptome_file = list(tm[tm["Reference Name"] == reference]["Location"])[0]           
+        # Write to files
+        chemistry_list.write("%s\t%s\n" % (sample, chemistry))
+        reference_list.write("%s\t%s\n" % (sample, reference))
+        transcriptome_list.write("%s\t%s\n" % (sample, transcriptome_file))
+    # close the files
+    transcriptome_list.close()
+    reference_list.close()
+    chemistry_list.close()
+    paths_list.close()
+
+def __main__(argv):
+    """Command Line parser for scRna-Seq pipeline.
+    Inputs- command line arguments
+    """
+    # the first command, -c, tells us which arguments we need to check for next, and which function we are going to call
```

Comment:
The approach entailed by this `-c` argument seems unconventional and rather limiting.  

For example:

```
wm10d-4e7:scripts eweitz$ python3 orchestra_methods.py --help
Error --help Is not a registered command
```

Should work.  

I'd suggest splitting this file into one library module (perhaps still named `orchestra_methods.py`) and one CLI module per command, e.g. `count.py`,  `mkfastq.py`, `analysis.py` and `parse.py`.  The goal of modularizing these commands is to make them easier to document and maintain.

Reply:
I'm not sure the individual commands are complex enough that splitting them into files is that helpful. Things like analysis and parse would only ever be used with each other during execution of the orchestra wdl. I moved everything to this file so the wdl can just call orchestra_methods.py-- every method relates to the orchestra wdl and sub wdls, which would never need to call --help.

Reply:
A working `--help` argument is important code documentation.  You're right that the WDLs would never need to call it.  It's often called by developers to get an overview of a new command-line interface.  All other Python scripts in this repo that use ArgParse support it, as do virtually all other command-line utilities I've used in Python, Perl, JavaScript, C++, etc.

The goal here is to get `python3 orchestra_methods.py --help` to work.  Modularizing code would be one hopefully straightforward way to do that, but if you can get `--help`to work with the current `-c` approach, then I'd defer to you there.

Reply:
Added

Reply:
The addition works for now; we should consider refactoring it down the line.

Manually writing two `help` entries for each argument -- one in prose [up top](https://github.com/broadinstitute/single_cell_portal/pull/24/files?utf8=%E2%9C%93&diff=split#diff-da8214dc67ec71f4f0e9ad8e3d36062cR2) and one in each `add_argument` call -- isn't quite what I had in mind.  It requires remembering to update both at the same time, and uses slightly unconventional formatting.  Using ArgParse's built-in `help` capabilities would be better, but incompatible with the `-c` argument approach.

But it is good that `--help` now returns something useful!  Thanks for the improvement.